sinatra-contrib/lib/sinatra/content_for.rb

```diff
@@ -128,9 +129,25 @@ module ContentFor
     #
     # Your blocks can also receive values, which are passed to them
     # by <tt>yield_content</tt>
-    def content_for(key, value = nil, &block)
+    def content_for(key, *args, &block)
+      args = args.dup.flatten
+      key = key.to_sym
+
+      value = if args.first.is_a?(String)
+                args.shift
+              else
+                nil
+              end
+
+      options = if args.last.is_a?(Hash)
+                  args.pop
+                else
+                  {}
+                end
+
       block ||= proc { |*| value }
-      content_blocks[key.to_sym] << capture_later(&block)
+      flush_key_set << key if options[:flush]
```

Comment:
I think this could be simplified to just something like:

```
clear_content_for(key) if options[:flush]
```

I'm almost certain that would take care of mimicking the Rails behavior.